src/Common/src/Interop/Unix/System.Globalization.Native/Interop.Idna.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class GlobalizationNative
+    {
+        internal const int AllowUnassigned = 0x1;
+        internal const int UseStd3AsciiRules = 0x2;
+
+        [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode)]
+        internal static extern int ToAscii(uint flags, string src, int srcLen, [Out] char[] dstBuffer, int dstBufferCapacity);
+
+        [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode)]
+        internal static extern int ToUnicode(uint flags, string src, int srcLen, [Out] char[] dstBuffer, int dstBufferCapacity);
```

Comment:
Yeah, you're right; with `char[]` it just pins the array without the need for any copying.